Framework/src/TrendingTask.cxx

```diff
@@ -310,75 +333,132 @@ std::string TrendingTask::deduceGraphLegendOptions(const TrendingTaskConfig::Gra
 TCanvas* TrendingTask::drawPlot(const TrendingTaskConfig::Plot& plotConfig)
 {
   auto* c = new TCanvas();
-  auto* legend = new TLegend(0.3, 0.2);
 
+  // Legend (NDC coordinates if enabled in config)
+  TLegend* legend = nullptr;
+  if (plotConfig.legend.enabled) {
+    legend = new TLegend(plotConfig.legend.x1, plotConfig.legend.y1,
+                         plotConfig.legend.x2, plotConfig.legend.y2,
+                         /*header=*/nullptr, /*option=*/"NDC");
+    if (plotConfig.legend.nColumns > 0) {
+      legend->SetNColumns(plotConfig.legend.nColumns);
+    }
+  } else {
+    legend = new TLegend(0.30, 0.20, 0.55, 0.35, nullptr, "NDC");
+  }
+  legend->SetBorderSize(0);
+  legend->SetFillStyle(0);
+  legend->SetTextSize(0.03);
+  legend->SetMargin(0.15);
+
+  // Keep palette behavior unless user forces explicit colors via per-graph style
   if (plotConfig.colorPalette != 0) {
-    // this will work just once until we bump ROOT to a version which contains this commit:
-    // https://github.com/root-project/root/commit/0acdbd5be80494cec98ff60ba9a73cfe70a9a57a
-    // and enable the commented out line
-    // perhaps JSROOT >7.7.1 will allow us to retain the palette as well.
     gStyle->SetPalette(plotConfig.colorPalette);
-    // This makes ROOT store the selected palette for each generated plot.
-    // TColor::DefinedColors(1); // TODO enable when available
   } else {
-    // we set the default palette
-    gStyle->SetPalette();
+    gStyle->SetPalette(); // default
   }
 
-  // regardless whether we draw a graph or a histogram, a histogram is always used by TTree::Draw to draw axes and title
-  // we attempt to keep it to do some modifications later
+  // Helpers
+  auto resolveColor = [](int idx) -> Color_t {
+    if (idx >= 0) {
+      return static_cast<Color_t>(idx);
+    }
+    return static_cast<Color_t>(-1);
+  };
+
+  auto getLastDrawnGraph = []() -> TGraph* {
+    if (!gPad) return nullptr;
+    TGraph* last = nullptr;
+    TIter it(gPad->GetListOfPrimitives());
+    while (TObject* obj = it()) {
+      if (obj->InheritsFrom(TGraph::Class())) {
+        last = static_cast<TGraph*>(obj);
+      }
+    }
+    return last;
+  };
+
+  auto applyStyleToGraph = [&](TGraph* gr, const TrendingTaskConfig::GraphStyle& st) {
```

Comment:
Please avoid [using abbreviations](https://rawgit.com/AliceO2Group/CodingGuidelines/master/naming_formatting.html?showone=General_Naming_Rules#General_Naming_Rules). Feel free to disagree, but names like `ln`, `mk`, `fl` are not obvious, so I don't see a room for exception here.